internal/graph/check.go

```diff
@@ -624,6 +651,56 @@ func (cc *ConcurrentChecker) checkComputedUserset(ctx context.Context, crc curre
 	return combineResultWithFoundResources(result, membershipSet)
 }
 
+// queryOptionsForArrowRelation returns query options such as SkipCaveats and SkipExpiration if *none* of the subject
+// types of the given relation support caveats or expiration.
+func (cc *ConcurrentChecker) queryOptionsForArrowRelation(ctx context.Context, reader datastore.Reader, namespaceName string, relationName string) ([]options.QueryOptionsOption, error) {
+	// TODO(jschorr): Change to use the type system once we wire it through Check dispatch.
+	nsDefs, err := reader.LookupNamespacesWithNames(ctx, []string{namespaceName})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(nsDefs) != 1 {
```

Comment:
what scenario does this represent? should it error?

Reply:
The namespace not being found, which *should* never happen, but if it does, better to just disable the optimization